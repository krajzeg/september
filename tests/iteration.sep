test_array := Array.test()

for := |sequence, body| {
	iterator := test_array.iterator()
	element := Nothing;
	
	# first element
	while(True) {
		try {
			element = iterator.next()
			body(element)
		} catch (ENoMoreElements) {
			break()
		}
	}
}

print("This should output numbers from 1 to 3.")
for (test_array, |x| {
	print(x);
})



