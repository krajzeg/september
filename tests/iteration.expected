This should output numbers from 1 to 3.
1
2
3
